cardano-cli/test/cardano-cli-golden/Test/Golden/Byron/Tx.hs

```diff
@@ -53,14 +52,22 @@ hprop_byronTx = propertyOnce $ H.moduleWorkspace "tmp" $ \tempDir -> do
 
   compareByronTxs createdTx expectedTx
 
-getTxByteString :: FilePath -> H.PropertyT IO (ATxAux ByteString)
+getTxByteString :: ()
+  => Member Hedgehog r
+  => Member (Embed IO) r
+  => FilePath
+  -> Sem r (ATxAux ByteString)
 getTxByteString txFp = do
-  eATxAuxBS <- liftIO . runExceptT $ readByronTx $ File txFp
-  case eATxAuxBS of
-    Left err -> failWith Nothing . docToString $ renderByronTxError err
-    Right aTxAuxBS -> return aTxAuxBS
+  embed (runExceptT $ readByronTx $ File txFp)
+    & onLeftM (failWith Nothing . docToString . renderByronTxError)
 
-compareByronTxs :: FilePath -> FilePath -> H.PropertyT IO ()
+compareByronTxs :: ()
+  => HasCallStack
+  => Member (Embed IO) r
+  => Member Hedgehog r
+  => FilePath
+  -> FilePath
+  -> Sem r ()
```

Comment:
Even though `getTxByteString` is not a test function we can use it seamlessly below.